app/javascript/components/patient/laboratory/Laboratory.js

```diff
@@ -44,29 +36,92 @@ class Laboratory extends React.Component {
     });
   };
 
+  toggleDeleteModal = () => {
+    let current = this.state.showDeleteModal;
+    this.setState({
+      showDeleteModal: !current,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleChange = event => {
```

Comment:
can you pull this method under the constructor? that way all the edit and delete stuff is together